src/sentry/sentry_metrics/visibility/metrics_blocking.py

```diff
@@ -0,0 +1,128 @@
+from dataclasses import dataclass
+from typing import Any, Dict, List, Mapping, Optional, Sequence, Set, TypedDict
+
+import sentry_sdk
+
+from sentry.models.project import Project
+from sentry.sentry_metrics.visibility.errors import MalformedBlockedMetricsPayloadError
+from sentry.utils import json
+
+BLOCKED_METRICS_PROJECT_OPTION_KEY = "sentry:blocked_metrics"
+
+
+class BlockedMetricsRelayConfig(TypedDict):
+    deniedNames: Sequence[str]
+
+
+@dataclass(frozen=True)
+class BlockedMetric:
+    metric_mri: str
+    tags: Set[str]
+
+    @classmethod
+    def from_dict(cls, dictionary: Mapping[str, Any]) -> Optional["BlockedMetric"]:
+        if "metric_mri" not in dictionary:
+            return None
+
+        return BlockedMetric(
+            metric_mri=dictionary["metric_mri"], tags=set(dictionary.get("tags") or [])
+        )
+
+    def merge(self, other: "BlockedMetric") -> "BlockedMetric":
+        return BlockedMetric(metric_mri=self.metric_mri, tags=self.tags.union(other.tags))
+
+    def to_dict(self) -> Mapping[str, Any]:
+        return self.__dict__
+
+
+@dataclass
+class BlockedMetrics:
+    metrics: List[BlockedMetric]
+
+    @classmethod
+    def load_from_project(cls, project: Project, repair: bool = False) -> "BlockedMetrics":
+        json_payload = project.get_option(BLOCKED_METRICS_PROJECT_OPTION_KEY)
+        if not json_payload:
+            return BlockedMetrics(metrics=[])
+
+        try:
+            blocked_metrics_payload = json.loads(json_payload)
+        except ValueError:
+            if repair:
```

Comment:
Repairing on write allows us to make sure we don't end up in a broken state of the system which requires a separate PR to unbreak.

Reply:
I do understand that this is a real possibility since anyone else could theoretically write to this field, but I do not see a value in trying to "repair" it by deleting the key completely. Raising an exception and opening an issue on Sentry would be enough IMO.

Reply:
I can remove this behavior, but if something is corrupted, we either delete it or figure out its content and make a PR to untangle it programmatically.

My main idea for doing so is that if a config of a user gets corrupted, they will now ingest all their blocked metrics and they don't have a way to block them again (they have to wait for our PR). With my system, they can immediately fix the problem with the downside being losing their previous settings. Still, it's a negligible problem since if the JSON is corrupted there likely won't be valuable data in the stored settings.